### DISCUSSION TITLE
Would UpSetJS work (non-interactively) with sets/combinations with empty `elems` arrays?

### DISCUSSION THREAD

**Discussion by bobular** [upsetjs/upsetjs#71] [Q&A] [answered]
Hi Samuel - great library - nice work!

In our work (for non-profit ClinEpiDB.org) we may encounter datasets that become too large to transmit to the client. We would compute the set and intersection sizes on the server.  Would UpSetJS work non-interactively and with no extensions (e.g. boxplots) if the `elems` of the sets/combinations were all empty arrays?

At some point we will hopefully try this anyway and post the answer here, but just wondering if you or someone in the community already knows the answer.

**Comment by sgratzl** [comment, marked as answer]
see also https://github.com/upsetjs/upsetjs/blob/main/examples/staticData/src/App.tsx

**Reply by bobular** [reply]
Oh, awesome.  My browsing of the code saw `IBaseSet.elems` as mandatory* - but I stand happily corrected - thank you!

(* nothing wrong with an empty array, in TypeScript terms, of course)

**Reply by sgratzl** [reply]
mandatory from an interface yes but it can be an empty list as long as the cardinality is properly set
